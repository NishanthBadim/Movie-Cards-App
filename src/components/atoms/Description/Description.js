import React from 'react';
import PropTypes from 'prop-types'


/**
- Use this component for providing description.
**/

const Description =(props) => {
    return(
        <p className="text-justify" style={{fontSize: '14px'}}>
            {props.description}
        </p>
    );
}

Description.propTypes= {
    /**
    Use the loading state to indicate that the data Avatar needs is still loading.
    */
    description: PropTypes.string,
}
export default Description;