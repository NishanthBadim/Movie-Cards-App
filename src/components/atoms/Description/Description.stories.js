import React from 'react'
import {withKnobs, text } from '@storybook/addon-knobs/react'
import Description from './Description'
import { ColorPalette } from '@storybook/components'

export default {
    component: Description,
    title: "Description",
    decorators: [withKnobs],
    parameters: {
        componentSubtitle: 'Gives the description of a movie'
    }
}

export const Default = () => {
    return (
        <Description description={text("text","Here goes the description of movie")} />
    )
}