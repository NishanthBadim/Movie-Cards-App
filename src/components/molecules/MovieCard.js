import React from 'react';
import StarRating from '../atoms/StarRating/StarRating';
import Rating from '../atoms/Rating/Rating';
import Poster from '../atoms/Poster/Poster';
import Title from '../atoms/Title/Title';
import Genre from '../atoms/Genre/Genre';
import Description from '../atoms/Description/Description';
import Release from '../atoms/Release/Release';
import '../../styles/components/movieCard.css';
import Knight from '../../images/Dark_Knight.jpg';
import Titanic from '../../images/Titanic.jpg';
import Inception from '../../images/Inception.jpg';
import Endgame from '../../images/Endgame.jpg';
import Infinity_War from '../../images/Infinity.jpg';
import Avatar from '../../images/Avatar.jpg';
import Interstellar from '../../images/Interstellar.jpg';
import Joker from '../../images/Joker.jpg';
import Parasite from '../../images/Parasite.jpg';
import Dont_Breathe from '../../images/DontBreathe.jpg';
import Furious from '../../images/Furious7.jpg'; 
import Legend from '../../images/Legend.jpg';


const MovieCard = (props) => {

    let image=urlName(props);
    return(
    <div className="movie-card">
        <div className="movie-card card">
            <Poster image={image} video={props.movie.videoId} />
            <div className="card-body">
                <Title title={props.movie.title} />
                <Genre genre={props.movie.genre} />
                <Release year={props.movie.year} />
                <Description description={props.movie.description} />
            </div>
            <div className="card-footer">
                    <div className="float-left mt-1">
                        <StarRating rating={props.movie.rating} />
                    </div>
                    <Rating rating={props.movie.rating} />
                </div>
            </div>
        </div>
    );
}

let urlName= (props) => {
    let name=props.movie.imageUrl;
    let image='';
    if(name==='Dark_Knight')
        image=Knight;
    else if(name==='Titanic')
        image=Titanic;
    else if(name==='Inception')
        image=Inception;
    else if(name==='Infinity_War')
        image=Infinity_War;
    else if(name==='Avatar')
        image=Avatar;
    else if(name==='Interstellar')
        image=Interstellar;
    else if(name==='Joker')
        image=Joker;
    else if(name==='Parasite')
        image=Parasite;
    else if(name==='Dont_Breathe')
        image=Dont_Breathe;
    else if(name==='Furious')
        image=Furious;
    else if(name==='Endgame')
            image=Endgame;
    else
        image=Legend;
    return image;
}

export default MovieCard;