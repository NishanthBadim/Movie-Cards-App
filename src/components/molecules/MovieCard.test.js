import React from 'react';
import {create} from 'react-test-renderer'
import "@testing-library/jest-dom/extend-expect";
import { render, screen, getByRole, fireEvent } from "@testing-library/react";
import MovieCard from './MovieCard'
import Description from '../atoms/Description/Description';


const movie1=
    {
        "id": 100,
        "title": "Avengers: Endgame",
        "genre": "Action/Sci-Fi",
        "description":"After Thanos, an intergalactic warlord, disintegrates half of the universe, the Avengers must reunite and assemble again to reinvigorate their trounced allies and restore balance.",
        "year": 2019,
        "imageUrl": "Endgame",
        "videoId": "hA6hldpSTF8",
        "rating": 4.9
    }
const movie2=

    {
        "id": 200,
        "title": "Joker",
        "genre": "Drama/Crime",
        "description": "Isolated, bullied and disregarded by society, Fleck begins a slow descent into madness as he transforms into the criminal mastermind known as the Joker.",
        "year": 2019,
        "imageUrl": "Joker",
        "videoId": "zAGVQLHvwOY",
        "rating": 4.7
    }
const movie3=
    {
        "id": 300,
        "title": "Parasite",
        "genre": "Thriller/Comedy",
        "description": "Greed and class discrimination threaten the newly formed symbiotic relationship between the wealthy Park family and the destitute Kim clan.",
        "year": 2019,
        "imageUrl": "Parasite",
        "videoId": "SEUXfv87Wpk",
        "rating": 4.7
    }
const movie4=
    {
        "id":400,
        "title": "Avengers: Infinity War",
        "genre": "Action/Sci-Fi",
        "description": "The Avengers must stop Thanos, an intergalactic warlord, from getting his hands on all the infinity stones. However, Thanos is prepared to go to any lengths to carry out his insane plan.",
        "year": 2018,
        "imageUrl":"Infinity_War",
        "videoId": "6ZfuNTqbHE8",
        "rating":4.4
    }
const movie5=
    {
        "id": 500,
        "title": "Don't Breathe",
        "genre": "Horror/Thriller",
        "description": "Three delinquents break into the house of Norman, who is blind, to steal his money. However, much to their horror, they discover that Norman is not as defenceless as he seems.",
        "year": 2016,
        "imageUrl": "Dont_Breathe",
        "videoId": "76yBTNDB6vU",
        "rating": 3.9
    }
const movie6={
    "id": 600,
    "title": "Furious 7",
    "genre": "Action/Thriller",
    "description": "Dominic and his family are caught in a quagmire when Shaw's brother seeks bloody revenge. They must save a crucial program from falling into the wrong hands.",
    "year": 2015,
    "imageUrl": "Furious",
    "videoId": "Skpu5HaVkOc",
    "rating": 4.1
}

jest.mock('../atoms/Title/Title',() => ({
    __esModule: true,
    default: () => { return <div></div>}
    }))
jest.mock('../atoms/Description/Description',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/Genre/Genre',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/Header/Header',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/Poster/Poster',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/Rating/Rating',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/Release/Release',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/SearchBar/SearchBar',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
jest.mock('../atoms/StarRating/StarRating',() => ({
    __esModule: true,
    default: () => { return <div></div>}
}))
test("Match the snapshot - MovieCard 1",() => {
    const input1=create(<MovieCard movie={movie1}/> );
    expect(input1.toJSON()).toMatchSnapshot();
    const input2=create(<MovieCard movie={movie2}/> );
    expect(input2.toJSON()).toMatchSnapshot();
    const input3=create(<MovieCard movie={movie3}/> );
    expect(input3.toJSON()).toMatchSnapshot();
    const input4=create(<MovieCard movie={movie4}/> );
    expect(input4.toJSON()).toMatchSnapshot();
    const input5=create(<MovieCard movie={movie5}/> );
    expect(input5.toJSON()).toMatchSnapshot();
    const input6=create(<MovieCard movie={movie6}/> );
    expect(input6.toJSON()).toMatchSnapshot();
})
/*
test("Image is present", () => {
    render(<MovieCard/>);
    const image=screen.getByAltText("movie-poster");
    expect(image).toBeInTheDocument();

})
test("Video is popping up", () => {
    const {getByRole, getByTestId} = render(<Poster/>)
    fireEvent.click(getByRole('button'))
    expect(getByTestId('modal')).toBeInTheDocument();
})
test("Button is present", () => {
    render(<Poster/>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
  });
*/