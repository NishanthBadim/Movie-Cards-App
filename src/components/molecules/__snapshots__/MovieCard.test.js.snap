// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - MovieCard 1 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 1 2`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 1 3`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 1 4`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 1 5`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 1 6`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 2 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 3 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 4 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 5 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Match the snapshot - MovieCard 6 1`] = `
<div
  className="movie-card"
>
  <div
    className="movie-card card"
  >
    <div />
    <div
      className="card-body"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="card-footer"
    >
      <div
        className="float-left mt-1"
      >
        <div />
      </div>
      <div />
    </div>
  </div>
</div>
`;
